test/tekton-tests.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script runs the integration tests with Tekton
+
+source $(dirname $0)/../vendor/knative.dev/test-infra/scripts/e2e-tests.sh
+source $(dirname $0)/e2e-common.sh
+
+# Add local dir to have access to built kn
+export PATH=$PATH:${REPO_ROOT_DIR}
+export TEKTON_VERSION=${TEKTON_VERSION:-v0.8.0}
+export E2E_PROJECT_ID=$(gcloud config get-value project)
+export GCR_REPO=gcr.io/${E2E_PROJECT_ID}/${E2E_BASE_NAME}-e2e-img/${RANDOM}
+export KN_E2E_NAMESPACE=tkn-kn
+
+# Script entry point.
+
+initialize $@
+
+header "Running integration tests for Tekton"
+
+# Install Tekton
+kubectl apply -f https://github.com/tektoncd/pipeline/releases/download/${TEKTON_VERSION}/release.yaml
+
+# Configure Docker so that we can create a secret for GCR
+gcloud auth configure-docker
```

Comment:
I believe these two lines will fail on Prow because docker initialization is slightly different (meaning you don't need to login). But only running this test on Prow will tell us exactly what's gonna happen... :/

Reply:
Won't work for google cloud

Reply:
```
WARNING: Your config file at [/root/.docker/config.json] contains these credential helper entries:
{
  "credHelpers": {
    "gcr.io": "gcr", 
    "us.gcr.io": "gcr", 
    "asia.gcr.io": "gcr", 
    "staging-k8s.gcr.io": "gcr", 
    "eu.gcr.io": "gcr"
  }
}
These will be overwritten.
Docker configuration file updated.
WARNING! Your password will be stored unencrypted in /root/.docker/config.json.
Configure a credential helper to remove this warning. See
https://docs.docker.com/engine/reference/commandline/login/#credentials-store
Login Succeeded
```
[logs](https://prow.knative.dev/view/gcs/knative-prow/pr-logs/pull/knative_client/528/pull-knative-client-integration-tests/1201497650701012992#1:build-log.txt%3A581) from earlier run of the PR using https://github.com/knative/client/pull/528/commits/8eb87ada028494b5569affc7a274abd6075a18fe#diff-0bff8fdbdcedf4cfbdcab24558df0df2R43

Reply:
LGTM for Prow. I'll let Doug resolve this if he's satisfied with the log above.

Reply:
if the user doesn't have `gcloud` then this will fail

Reply:
Yeah. This script is mostly written for CI/Prow. It will need more changes in order to support non-CI and non-GCR registries.